crypto/tbs/src/lib.rs

```diff
@@ -174,40 +175,49 @@ pub fn sign_blinded_msg(msg: BlindedMessage, sks: SecretKeyShare) -> BlindedSign
     BlindedSignatureShare(sig.to_affine())
 }
 
-/// Combines a sufficient amount of valid blinded signature shares to a blinded
-/// signature. The responsibility of verifying the supplied shares lies with the
-/// caller.
-///
-/// * `sig_shares`: an iterator yielding pairs of key indices and signature
-///   shares from said key
-/// * `threshold`: number of shares needed to combine a signature
-///
+/// Combines the exact threshold of valid blinded signature shares to a blinded
+/// signature. The responsibility of verifying the shares lies and supplying
+/// exactly the necessary threshold of shares lies with the caller.
 /// # Panics
-/// If the amount of shares supplied is less than the necessary amount
-pub fn combine_valid_shares<I>(sig_shares: I, threshold: usize) -> BlindedSignature
-where
-    I: IntoIterator<Item = (usize, BlindedSignatureShare)>,
-    I::IntoIter: Clone + ExactSizeIterator,
-{
-    let points = sig_shares
-        .into_iter()
-        .take(threshold)
-        .map(|(idx, share)| {
-            let x = Scalar::from((idx as u64) + 1);
-            let y = share.0.into();
-            (x, y)
-        })
-        .collect::<Vec<(Scalar, G1Projective)>>();
-    if points.len() < threshold {
-        panic!("Not enough signature shares");
-    }
+/// If the amount of shares supplied is less than two
+pub fn aggregate_signature_shares(
```

Comment:
Should likely not be pub as it's not safe to use for non-experts. This function assumes that:
* Supplied shares are valid
* The correct number of shares is supplied (unintuitively, supplying too many will also lead to wrong results)